src/crypto/algorithms/megolm.ts

```diff
@@ -591,8 +591,9 @@ class MegolmEncryption extends EncryptionAlgorithm {
         userDeviceMap: IOlmDevice[],
         payload: IPayload,
     ): Promise<void> {
-        const contentMap: Record<string, Record<string, IEncryptedContent>> = {};
-        const deviceInfoByDeviceId = new Map<string, DeviceInfo>();
+        const userDeviceMapping: Record<string, Record<string, {
+            encryptedContent: IEncryptedContent; deviceInfo: DeviceInfo;
+        }>> = {};
```

Comment:
Could we use the alternate format which allows identifiers?

```suggestion
        const userDeviceMapping: {
            [userId: string]: {
                [deviceId: string]: {
                    encryptedContent: IEncryptedContent;
                    deviceInfo: DeviceInfo;
                };
            };
        } = {};
```